src/App.tsx

```diff
@@ -1,10 +1,58 @@
-import "./App.css";
+import styles from "./App.module.css";
+import Form from "./components/common/Form/Form";
+import { useState } from "react";
+import ShelfHeader from "./components/ShelfHeader/ShelfHeader";
+import { CARD_NUMBER_UNIT_LENGTH } from "./constants";
+import CardNumbersInputField from "./components/Field/CardNumbersInputField/CardNumbersInputField";
+import CardExpirationDateInputField from "./components/Field/CardExpirationDateInputField/CardExpirationDateInputField";
+import CardOwnerNameInputField from "./components/Field/CardOwnerNameInputField/CardOwnerNameInputField";
+import Card from "./components/Card/Card";
 
 function App() {
+  const [cardNumbers, setCardNumbers] = useState<string[]>(
+    new Array(CARD_NUMBER_UNIT_LENGTH).fill("")
+  );
```

Comment:
(마이너)
new Array의 결과물이 의미 있게 사용되는 건 첫 렌더일 때 같은데 지금은 매번 컴포넌트가 호출될 때마다 생성자가 호출될 것 같습니다.

```suggestion
  const [cardNumbers, setCardNumbers] = useState<string[]>(
    () => new Array(CARD_NUMBER_UNIT_LENGTH).fill("")
  );
```

Reply:
함수형으로 선언하면 초기 한 번만 생성자가 호출되는군요!

좋은 정보 감사합니다 😄